Add Epinio installation on Rancher (#44)
epinio/docs: src/installation/install_epinio_on_rancher.md

# Rancher configuration

Before installing Epinio, you need a running Rancher instance.

> If you don't get it installed yet, feel free to follow installation guides [here](https://rancher.com/docs).

Reviewer: ```suggestion
If you do not have such, please use the Rancher's [installation guides](https://rancher.com/docs) to set one up.
```

## Epinio Prerequisites

### Kubernetes cluster

> The supported configuration is to either deploy a new cluster or import an existing one in Rancher (those steps are detailed in the Rancher docs).</br>

Reviewer: ```suggestion
The two supported methods are to either deploy a new cluster or to import an existing one into Rancher (those steps are detailed in the Rancher docs).</br>
```

Is the reference to detailed steps for deploying, for importing, or both ? (The phrasing above is ambiguous to me. Recommend splitting that part of into its own sentence).

Author: Deploying or importing are explained [here](https://rancher.com/docs/rancher/v2.6/en/cluster-provisioning/).
I can mention it but it will stick to v2.6.

Reviewer: Ok. So the reference is for both methods. I change my proposal to:
```suggestion
The two supported methods are to either deploy a new cluster or to import an existing one into Rancher. The steps for take for either method are detailed in the Rancher documentation (See `Cluster Provisioning`).</br>
```

> *For development purpose, you can imagine deploying Epinio on the same node where Rancher is installed but it requires more steps and keep in mind it's not supported by Rancher.*

Reviewer: ```suggestion
*While it is possible (for development) to deploy Epinio on the same node where Rancher is installed, this requires more steps and is not supported by Rancher either.*
```

### Rancher settings
At the time of writing, Epinio is not included in the latest stable Rancher version.</br>

> This means you need to update two settings in your Rancher dashboard.</br>

Reviewer: ```suggestion
This means that you have to update two settings in your Rancher dashboard.</br>
```

Go to `Global settings` > `Advanced settings`  and update the following settings:</br>

| | |
|--|--|
| `ui-dashboard-index`  | `https://releases.rancher.com/dashboard/epinio-v0.3.0/index.html` |
| `ui-offline-preferred`  | `Remote` |

> *Please, keep in mind that `ui-dashboard-index` value will change over the time until it is not merged in stable Rancher version.*

Reviewer: I have trouble understanding this sentence. Does it mean that the setting will be removed from Rancher in some future release ?

Author: Yes exactly, Epinio's UI code is not included yet in the Rancher code.
However, maybe I can mention versions of all components I used at the beginning of the document.

Reviewer: Ok. So I managed to pull the correct meaning out of this. While I am still convinced that it could be phrased better, I also have no idea how to rephrase it. Maybe @agracey can help ? Lets ignore this for now.

## Install Epinio

In the Rancher Dashboard, go to the cluster on where you want to install Epinio.

Add the Epinio Helm repo in `Apps & Marketplace` >  `Repositories` > `Create`

Give a name to the repo and use `https://epinio.github.io/helm-charts` as index URL.

> Once done, go to Chart menu and install the `epinio-installer` chart.

Reviewer: ```suggestion
Once done, go to the Chart menu and install the `epinio-installer` chart.
```

> You can select the version you want but most of the time you will choose the latest one (default), then click install in the top right corner.

Reviewer: ```suggestion
While you can select the version you want most of the time you should choose the latest one (default). Then click install in the top right corner.
```

> On the next screen, you can select a namespace where the deployment will happen and you have to set a name to the deployment.

Reviewer: ```suggestion
On the next screen, you can select a namespace where the deployment will happen. You also have to set a name for the deployment.
```

Question: `can select`, or `have to select` ? If `can`, what does happen if no namespace is selected ? (Or, in other words, what is the default namespace ?)

Author: Namespace default is automatically selected. So it's not possible to let namespace empty there.
That's why I use `can` because you can keep default or select another namespace.

Reviewer: Ok. That answers my question. No need to redo my change proposal.

> Next screen is where you can configure Epinio according to your needs.</br>

Reviewer: ```suggestion
The following screen is where you can configure Epinio according to your needs.</br>
```

> One of the most important setting here is `Access control allow origin`, it's mandatory filling your Rancher dashboard URL there.</br>

Reviewer: ```suggestion
The most important setting here is `Access control allow origin`. It is mandatory to insert your Rancher dashboard URL into this field.</br>
```

It basically allows Rancher to talk with the Epinio API.

> Once everything configured, click on the install button and wait few minutes for the installation to complete.</br>

Reviewer: ```suggestion
Once everything is configured, click on the install button and then wait a few minutes for the installation to complete.</br>
```

> Follow instructions given by the Helm install command output.

Reviewer: ```suggestion
After that follow the instructions given by the Helm install command output.
```

## Access the Epinio menu

Click on the Epinio icon in the main side menu (under **GLOBAL APPS**).

You should see your fresh Epinio instance there.

> *Pay attention, if you use an untrusted certificate, you have to click on the URL, accept the security exception and use the refresh button below Epinio instances.</br>

Reviewer: ```suggestion
__Attention__. If you use an untrusted certificate, you have to click on the URL, accept the security exception and use the refresh button shown below Epinio instances.</br>
```

I am tempted to write `shown below the table of Epinio instances`, however I do not know enough about the GUI to be sure if that is true or not.

Author: Actually, the refresh button is below page title `Epinio instances` and not below the table.
![Screenshot from 2021-11-15 11-20-54](https://user-images.githubusercontent.com/6025636/150305396-4e23a941-26bd-4d35-81a5-a09ab8479faf.png)

Reviewer: Ah. Ok, changing the proposal to:
```suggestion
__Attention__. If you use an untrusted certificate, you have to click on the URL, accept the security exception and use the refresh button shown below the page title, i.e. __Epinio instances__.</br>
```

# Rancher configuration

Before installing Epinio, you need a running Rancher instance.

If you don't get it installed yet, feel free to follow installation guides [here](https://rancher.com/docs).

## Epinio Prerequisites

### Kubernetes cluster

> For development purpose, you can deploy Epinio on the same node where Rancher installed but it's not supported for production.</br>

Author: I think it's fine to mention the possibility of doing it but I changed to mention the supported way first.

That's why first, we are about to detail how to install Epinio in a separate Kubernetes cluster.</br>
In this case, you need to either deploy a new cluster or import an existing one in Rancher. Those steps are detailed in the Rancher docs.

### Rancher settings
At the time of this documentation is written, Epinio is not included in the latest stable Rancher version.</br>

> That means you need to update two settings in your Rancher dashboard.</br>

Reviewer 2: MAybe better to use `This` instead of `That`? I checked [here](https://keydifferences.com/difference-between-this-and-that.html#:~:text=The%20words%20'this'%20and%20',is%20used%20for%20indicating%20something.&text=We%20use%20the%20word%20'this,which%20is%20farther%20from%20you.) but I'm still confused :D.

In the Rancher Dashboard, go to the cluster on where you want to install Epinio.

Add the Epinio Helm repo in `Apps & Marketplace` >  `Repositories` > `Create`

Give a name to the repo and use `https://epinio.github.io/helm-charts` as index URL.

Once done, go to Chart menu and install the `epinio-installer` chart.

You can select the version you want but most of the time you will choose the latest one (default), then click install in the top right corner.

On the next screen, you can select a namespace where the deployment will happen and you have to set a name to the deployment.

Next screen is where you can configure Epinio according to your needs.</br>
One of the most important setting here is `Access control allow origin`, it's mandatory filling your Rancher dashboard URL there.</br>
It basically allows Rancher to talk with the Epinio API.

Once everything configured, click on the install button and wait few minutes for the installation to complete.</br>
Follow instructions given by the Helm install command output.

## Access the Epinio menu

Click on the Epinio icon in the main side menu (under **GLOBAL APPS**).

You should see your fresh Epinio instance there.

> *Pay attention, if you use an invalid certificate, you have to click on the URL, accept the security exception and use the refresh button below Epinio instances.</br>

Reviewer 2: Regarding the discussion we had on `invalid`, I think that in fact we can directly write `untrusted` here, WDYT?
